visvis2/material/_line.py

```diff
@@ -1,6 +1,32 @@
-from visvis2.material._base import Material
+from python_shader import Struct, vec3, f32
+
+from ._base import Material
+from ..utils import array_from_shadertype
+from .. import BufferWrapper
 
 
 class LineStripMaterial(Material):
-    def __init__(self):
+
+    uniform_type = Struct(color=vec3, thickness=f32)
+
+    def __init__(self, color=(1, 1, 1), thickness=2.0):
         super().__init__()
+
+        array = array_from_shadertype(self.uniform_type)
+        self.uniforms = BufferWrapper(array, usage="uniform", mapped=True)
+        self.set_color(color)
+        self.set_thickness(thickness)
```

Comment:
:+1: material is just a type and a bag of parameters mapped to uniforms
:-1: the types come from python-shader

Reply:
It's an implementation detail that we use that type system to specify the structure of the uniforms ;)

Reply:
I feel your point though ... I'll make an issue.

Reply:
#21